x-pack/plugins/ml/server/lib/alerts/alerting_service.ts

```diff
@@ -257,14 +266,31 @@ export function alertingServiceProvider(mlClient: MlClient) {
     };
   };
 
+  /**
+   * Provides unique key for the anomaly result.
+   */
+  const getAlertInstanceKey = (source: any): string => {
+    let alertInstanceKey = `${source.job_id}_${source.timestamp}`;
+    if (source.result_type === ANOMALY_RESULT_TYPE.INFLUENCER) {
+      alertInstanceKey += `_${source.influencer_field_name}_${source.influencer_field_value}`;
+    } else if (source.result_type === ANOMALY_RESULT_TYPE.RECORD) {
+      const fieldName = getEntityFieldName(source);
+      const fieldValue = getEntityFieldValue(source);
+      alertInstanceKey += `_${source.detector_index}_${source.function}_${fieldName}_${fieldValue}`;
```

Comment:
`source.detector_index` is `undefined` in the key my test generated. Is this available in the `source` ?

Reply:
forgot to include it into the source, fixed in https://github.com/elastic/kibana/pull/91485/commits/61764c5ba3ea89afd5aecfb7fde636f7a6da923c